cli/js/dispatch_minimal.ts

```diff
@@ -17,23 +21,57 @@ export interface RecordMinimal {
   opId: number; // Maybe better called dispatchId
   arg: number;
   result: number;
+  err?: {
+    kind: ErrorKind;
+    message: string;
+  };
+}
+
+export function readInt32LittleEndian(
+  buffer: Uint8Array,
+  offset: number
+): number {
+  return (
+    (buffer[offset + 3] << 24) |
+    (buffer[offset + 2] << 16) |
+    (buffer[offset + 1] << 8) |
+    buffer[offset]
+  );
 }
 
 export function recordFromBufMinimal(
   opId: number,
-  buf32: Int32Array
+  ui8: Uint8Array
 ): RecordMinimal {
-  if (buf32.length != 3) {
-    throw Error("Bad message");
+  const promiseId = readInt32LittleEndian(ui8, 0);
+  const arg = readInt32LittleEndian(ui8, 4);
+  const result = readInt32LittleEndian(ui8, 8);
```

Comment:
This is likely too much work to decode. Converting to Int32Array and reading elements is faster.